maestro-client/src/main/java/maestro/drivers/WebDriver.kt

```diff
@@ -203,6 +208,20 @@ class WebDriver(val isStudio: Boolean) : Driver {
         return parse(contentDesc as Map<String, Any>)
     }
 
+    private fun detectWindowChange() {
+        // Checks whether there are any new window handles available and, if so, switches Selenium driver focus to it
+        val driver = ensureOpen()
+
+        if (lastSeenWindowHandles != driver.windowHandles) {
+            val newHandles = driver.windowHandles - lastSeenWindowHandles
+            lastSeenWindowHandles = driver.windowHandles
+
+            if (newHandles.isNotEmpty()) {
+                driver.switchTo().window(newHandles.first())
```

Comment:
Could we log something here so it is clear that selenium automatically shifted focus to a new tab?